compat/GnuRegex.c

```diff
@@ -40,6 +40,12 @@
 
 #if USE_GNUREGEX /* only if squid needs it. Usually not */
 
+/* recent GCC 12.1 does not like array magic done here */
+#if __GNUC__ >= 12 && __GNUC_MINOR__ >= 1
```

Comment:
We should not skip GCC v13.0, v14.0, etc.. Also, we should explain why we need to be fairly precise here.

Finally, let's not assert that this is only about array magic.

```suggestion
/* Starting with v12.1, GCC warns of various problems with this ancient code. */
/* GCC versions prior to v12.1 do not support these pragmas. */
#if (__GNUC__ == 12 && __GNUC_MINOR__ >= 1) || (__GNUC__ > 12)
```